initialization/initialization.go

```diff
@@ -325,22 +334,32 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 	return fmt.Errorf("no nonce found")
 }
 
-func (init *Initializer) removeRedundantFiles(layout filesLayout) error {
-	numFiles, err := init.diskState.NumFilesWritten()
+func removeRedundantFiles(cfg config.Config, opts config.InitOpts, logger *zap.Logger) error {
```

Comment:
Why not leave it as method? you can still call it in tests if its private.

Reply:
To simplify testing - to avoid creating an instance of Initializer just to test this logic.